### DISCUSSION TITLE
sheet.saveUpdatedCells() doesn't seem to save any data

### DISCUSSION THREAD

**Discussion by Tristan-zimmerman** [theoephraim/node-google-spreadsheet#549] [Q&A] [answered]
I've been importing data from Google Sheets to our database and writing the newly generated record IDs back to the spreadsheet.

If I use `rows[index].save()` then the data appears no problem (though I've hit the API limit a few times). If I use `sheet.saveUpdatedCells()` after i've added the data to the row then nothing seems to happen at all.

I'm using the service account auth process and note the saveUpdatedCells call is in two different places in the example (this is just for the example) as I've tried it outside of the `Promise.allSettled` call as well as the in the last `then` from the `Promise.allSettled` call.

Here's my code example:
```Javascript
const pushDataToDb = async (productData) => {
    Promise.allSettled(productData.map(async (product, index) => {
      const endpoint = product.id
        ? `${apiDomain}/product/${product.id.trim()}`
        : `${apiDomain}/product`

      await fetch(
        endpoint,
        {
          method: product.id ? 'PUT' : 'POST',
          mode: 'cors',
          cache: 'no-cache',
          credentials: 'same-origin',
          headers: {
            'Content-Type': 'application/json'
          },
          body: JSON.stringify(product)
        }
      ).then(response => {
        return response.json()
      })
        .then(async (data) => {
          if (product.id) {
            rows[index].id = data.id
            await rows[index].save()
          }
          console.log(`
            id: ${data.id},
            productName: ${data.productName}
          `)
        })
    })).then(async (results) => {
      console.log('cool: ', results)
      sheet.saveUpdatedCells()
    })

    await sheet.saveUpdatedCells()
}
```

Any help would be appreciated! Thanks in advance!

**Comment by theoephraim** [comment, marked as answer]
The row-based and cell-based APIs are totally separate...

If you want to deal with rows (which keeps this simple but is limited) then you load rows, update them (while being blissfully unaware of the concept of cells), and call save on them.

Otherwise if you want to deal with cells directly (more flexible but more complex), then you load a range of cells, update the data in specific cells, and then can save all the updates.

This was set up this way to try to mirror Google's row-based interactions that they used to have in a previous version  of their API. It's definitely much more limited, but probably handles 90% of use cases and keeps things much simpler.
